### ISSUE TITLE
Icinga 2.11.0 RC1: Problem connecting to 2.10.5

### ISSUE THREAD

**Issue by peteeckel** [Icinga/icinga2#7366]
I just upgraded my test environment to 2.11.0 RC1 on the master servers and some satellites and agents, while staying at version 2.10.5 on one satellite cluster. All systems are CentOS 7.6 freshly installed from the latest bento Vagrant box 201907.24.0.

The 2.11.0 systems came up and connected just fine, but the 2.10.5 satellites fail to connect to the 2.11.0 master with the following error message in the log:

Satellites:
```
[2019-07-27 13:18:30 +0000] warning/TlsStream: OpenSSL error: error:1408A0C1:SSL routines:ssl3_get_client_hello:no shared cipher
[2019-07-27 13:18:30 +0000] critical/ApiListener: Client TLS handshake failed (from [192.168.106.102]:37838): Error: Socket was closed during TLS handshake.

        (0) Handling new API client connection

Context:
        (0) Handling new API client connection
```

Master:
```
[2019-07-27 13:19:00 +0000] critical/ApiListener: Client TLS handshake failed (to [192.168.106.114]:5665): sslv3 alert handshake failure
[2019-07-27 13:19:00 +0000] information/ApiListener: Finished reconnecting to endpoint 'satellite4.dev.hindenburgring.com' via host 'satellite4.dev.hindenburgring.com' and port '5665'
```

The OpenSSL version on both ends is recent as well:
```
openssl.x86_64                1:1.0.2k-16.el7_6.1      @updates                 
openssl-libs.x86_64           1:1.0.2k-16.el7_6.1      @updates                 
```

I ran `sslscan` against both the master and the satellite:
```
lagavulin:monitoring pete$ sslscan master1.dev.hindenburgring.com:5665
[...]
  Supported Server Cipher(s):
Preferred TLSv1.2  256 bits  ECDHE-RSA-AES256-GCM-SHA384   Curve P-256 DHE 256
Accepted  TLSv1.2  128 bits  ECDHE-RSA-AES128-GCM-SHA256   Curve P-256 DHE 256
Accepted  TLSv1.2  256 bits  ECDHE-RSA-AES256-SHA384       Curve P-256 DHE 256
Accepted  TLSv1.2  128 bits  ECDHE-RSA-AES128-SHA256       Curve P-256 DHE 256
[...]
```

```
lagavulin:monitoring pete$ sslscan satellite3.dev.hindenburgring.com:5665
[...]
Supported Server Cipher(s):
Preferred TLSv1.2  256 bits  AES256-GCM-SHA384            
Accepted  TLSv1.2  256 bits  AES256-SHA256                
Accepted  TLSv1.2  256 bits  AES256-SHA                   
Accepted  TLSv1.2  256 bits  CAMELLIA256-SHA              
Accepted  TLSv1.2  128 bits  AES128-GCM-SHA256            
Accepted  TLSv1.2  128 bits  AES128-SHA256                
Accepted  TLSv1.2  128 bits  AES128-SHA                   
Accepted  TLSv1.2  128 bits  CAMELLIA128-SHA              
Preferred TLSv1.1  256 bits  AES256-SHA                   
Accepted  TLSv1.1  256 bits  CAMELLIA256-SHA              
Accepted  TLSv1.1  128 bits  AES128-SHA                   
Accepted  TLSv1.1  128 bits  CAMELLIA128-SHA              
Preferred TLSv1.0  256 bits  AES256-SHA                   
Accepted  TLSv1.0  256 bits  CAMELLIA256-SHA              
Accepted  TLSv1.0  128 bits  AES128-SHA                   
Accepted  TLSv1.0  128 bits  CAMELLIA128-SHA              
[...]
```

And indeed, the cipher lists do not match.

The API objects are rather plain, no TLS or cipher restrictions apply:
```
object ApiListener "api" {
    bind_host = "master1.dev.hindenburgring.com"

    ticket_salt = TicketSalt

    accept_config   = true
    accept_commands = true
}
```

```
object ApiListener "api" {
    bind_host = "satellite3.dev.hindenburgring.com"

    accept_config   = true
    accept_commands = true
}
```

Version info:
```
[root@master1 icinga2]# icinga2 --version
icinga2 - The Icinga 2 network monitoring daemon (version: 2.11.0-0.rc1.1)

Copyright (c) 2012-2019 Icinga GmbH (https://icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

System information:
  Platform: CentOS Linux
  Platform version: 7 (Core)
  Kernel: Linux
  Kernel version: 3.10.0-957.21.3.el7.x86_64
  Architecture: x86_64

Build information:
  Compiler: GNU 4.8.5
  Build host: runner-LTrJQZ9N-project-322-concurrent-0

Application information:

General paths:
  Config directory: /etc/icinga2
  Data directory: /var/lib/icinga2
  Log directory: /var/log/icinga2
  Cache directory: /var/cache/icinga2
  Spool directory: /var/spool/icinga2
  Run directory: /run/icinga2

Old paths (deprecated):
  Installation root: /usr
  Sysconf directory: /etc
  Run directory (base): /run
  Local state directory: /var

Internal paths:
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid
```

```
[root@satellite3 ~]# icinga2 --version
icinga2 - The Icinga 2 network monitoring daemon (version: r2.10.5-1)

Copyright (c) 2012-2019 Icinga GmbH (https://icinga.com/)
License GPLv2+: GNU GPL version 2 or later <http://gnu.org/licenses/gpl2.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

System information:
  Platform: CentOS Linux
  Platform version: 7 (Core)
  Kernel: Linux
  Kernel version: 3.10.0-957.21.3.el7.x86_64
  Architecture: x86_64

Build information:
  Compiler: GNU 4.8.5
  Build host: afcad697467e

Application information:

General paths:
  Config directory: /etc/icinga2
  Data directory: /var/lib/icinga2
  Log directory: /var/log/icinga2
  Cache directory: /var/cache/icinga2
  Spool directory: /var/spool/icinga2
  Run directory: /run/icinga2

Old paths (deprecated):
  Installation root: /usr
  Sysconf directory: /etc
  Run directory (base): /run
  Local state directory: /var

Internal paths:
  Package data directory: /usr/share/icinga2
  State path: /var/lib/icinga2/icinga2.state
  Modified attributes path: /var/lib/icinga2/modified-attributes.conf
  Objects path: /var/cache/icinga2/icinga2.debug
  Vars path: /var/cache/icinga2/icinga2.vars
  PID path: /run/icinga2/icinga2.pid
```

Upgrading the satellites in question to 2.11.0 rc1 fixed the problem immediately without any further action, so it is probably not related to the general setup, certificates etc. in place (the setup remained exactly the same except for the updated Icinga 2/BOOST versions).

**Comment by peteeckel** [comment]
I ran another test against a 2.10.5 satellite, checking the TLS ciphers offered by the satellite (edited for readability):

```
openssl s_server -accept 5665 -key [...]
[...]
Shared ciphers:
ECDHE-RSA-AES256-GCM-SHA384
ECDHE-ECDSA-AES256-GCM-SHA384
ECDHE-RSA-AES256-SHA384
ECDHE-ECDSA-AES256-SHA384
ECDHE-RSA-AES256-SHA
ECDHE-ECDSA-AES256-SHA
DH-DSS-AES256-GCM-SHA384
DHE-DSS-AES256-GCM-SHA384
DH-RSA-AES256-GCM-SHA384
DHE-RSA-AES256-GCM-SHA384
DHE-RSA-AES256-SHA256
DHE-DSS-AES256-SHA256
DH-RSA-AES256-SHA256
DH-DSS-AES256-SHA256
DHE-RSA-AES256-SHA
DHE-DSS-AES256-SHA
DH-RSA-AES256-SHA
DH-DSS-AES256-SHA
DHE-RSA-CAMELLIA256-SHA
DHE-DSS-CAMELLIA256-SHA
DH-RSA-CAMELLIA256-SHA
DH-DSS-CAMELLIA256-SHA
ECDH-RSA-AES256-GCM-SHA384
ECDH-ECDSA-AES256-GCM-SHA384
ECDH-RSA-AES256-SHA384
ECDH-ECDSA-AES256-SHA384
ECDH-RSA-AES256-SHA
ECDH-ECDSA-AES256-SHA
AES256-GCM-SHA384
AES256-SHA256
AES256-SHA
CAMELLIA256-SHA
ECDHE-RSA-AES128-GCM-SHA256
ECDHE-ECDSA-AES128-GCM-SHA256
ECDHE-RSA-AES128-SHA256
ECDHE-ECDSA-AES128-SHA256
ECDHE-RSA-AES128-SHA
ECDHE-ECDSA-AES128-SHA
DH-DSS-AES128-GCM-SHA256
DHE-DSS-AES128-GCM-SHA256
DH-RSA-AES128-GCM-SHA256
DHE-RSA-AES128-GCM-SHA256
DHE-RSA-AES128-SHA256
DHE-DSS-AES128-SHA256
DH-RSA-AES128-SHA256
DH-DSS-AES128-SHA256
DHE-RSA-AES128-SHA
DHE-DSS-AES128-SHA
DH-RSA-AES128-SHA
DH-DSS-AES128-SHA
DHE-RSA-SEED-SHA
DHE-DSS-SEED-SHA
DH-RSA-SEED-SHA
DH-DSS-SEED-SHA
DHE-RSA-CAMELLIA128-SHA
DHE-DSS-CAMELLIA128-SHA
DH-RSA-CAMELLIA128-SHA
DH-DSS-CAMELLIA128-SHA
ECDH-RSA-AES128-GCM-SHA256
ECDH-ECDSA-AES128-GCM-SHA256
ECDH-RSA-AES128-SHA256
ECDH-ECDSA-AES128-SHA256
ECDH-RSA-AES128-SHA
ECDH-ECDSA-AES128-SHA
AES128-GCM-SHA256
AES128-SHA256
AES128-SHA
SEED-SHA
CAMELLIA128-SHA
ECDHE-RSA-DES-CBC3-SHA
ECDHE-ECDSA-DES-CBC3-SHA
EDH-RSA-DES-CBC3-SHA
EDH-DSS-DES-CBC3-SHA
DH-RSA-DES-CBC3-SHA
DH-DSS-DES-CBC3-SHA
ECDH-RSA-DES-CBC3-SHA
ECDH-ECDSA-DES-CBC3-SHA
DES-CBC3-SHA
IDEA-CBC-SHA
ECDHE-RSA-RC4-SHA
ECDHE-ECDSA-RC4-SHA
ECDH-RSA-RC4-SHA
ECDH-ECDSA-RC4-SHA
RC4-SHA
RC4-MD5
```

Obviously the 2.10.5 satellite **does** offer, e.g. `ECDHE-RSA-AES256-GCM-SHA384`, but the master does not accept/recognise the offered cipher.

**Comment by peteeckel** [comment]
Another update (and an explanation for the former finding):

The specific satellite is configured to only accept connections from the master server, and to not try connection to the master server itself. So the ciphers offered **to** the master do not matter, as there is no connection attempt done by the satellite. 

I reconfigured this for bidirectional connection and checked the ciphers offered by the master to the satellite:
```
[root@satellite3 ~]# openssl s_server -accept 5665 -key [...]
[...]
Shared ciphers:
ECDHE-ECDSA-AES256-GCM-SHA384
ECDHE-RSA-AES256-GCM-SHA384
ECDHE-ECDSA-AES128-GCM-SHA256
ECDHE-RSA-AES128-GCM-SHA256
ECDHE-ECDSA-AES256-SHA384
ECDHE-RSA-AES256-SHA384
ECDHE-ECDSA-AES128-SHA256
ECDHE-RSA-AES128-SHA256
```

So obviously the master not only does not accept anything but ECDHE ciphers, it also does not offer anything else. Which explains why it cannot connect to the 2.10.5 satellite.

The situation is now as follows: 

* The 2.10.5 satellite does not offer the ECDHE ciphers as a server (confirmed by `sslscan`)
* The 2.11.0 RC1 master only offers ECDHE ciphers as a server (confirmed by `sslscan`)
* The 2.10.5 satellite offers ECDHE ciphers as well as other ciphers as a client (confirmed by `openssl s_server`)
* The 2.11.0 RC1 master only offers ECDHE ciphers as a client (confirmed by `openssl s_server`)

As a result, a 2.11.0 RC1 Icinga 2 client **cannot** connect to a 2.10.5 server, but a 2.10.5 client **can** connect to a 2.11.0 RC1 server.

**Comment by peteeckel** [comment]
I just edited the subject (again) to make it clear that the problem is not related to master/satellite connections but can be reproduced with any connection between a 2.11.0 RC1 and 2.10.5 Icinga instance.

As long as bidirectional connections are allowed, the problem is obscured by the fact that eventually the icinga 2.10.5 daemon will connect to the 2.11.0 RC1 daemon.

**Comment by dnsmichi** [comment]
> The specific satellite is configured to only accept connections from the master server, and to not try connecting to the master server itself. So the ciphers offered to the master do not matter, as there is no connection attempt done by the satellite.

Ok, that's it. The satellite being the `server` role prefers the cipher, and there's none.

Then #7247 still bites us @lippserd 

I see two options:

- soften the cipher list so that el7 works again
- release 2.10.6, 2.9.4 and fix the el7 bug we've been carrying around for a while

**Comment by peteeckel** [comment]
I would advocate the solution of at least supporting some additional ciphers for outgoing connections or - maybe the most elegant solution - making this configurable with a cipher_list option in the Endpoint object  for outgoing connections analogous to the cipher_list option in the ApiListener object for incoming connections.

**Comment by dnsmichi** [comment]
# Default Cipher Suites

## 2.10

```
michi@Michaels-MacBook-Pro ~/dev/icinga/icinga2 (bugfix/ecdh-ciphers-el7-2-10=) $ grep -A 2 cipher_list lib/remote/apilistener.ti
	[config] String cipher_list {
		default {{{ return "ALL:!LOW:!WEAK:!MEDIUM:!EXP:!NULL"; }}}
	};
```

## 2.11 RC1

```
michi@Michaels-MacBook-Pro ~/dev/icinga/icinga2 (master=) $ grep -A 2 cipher_list lib/remote/apilistener.ti
	[config] String cipher_list {
		default {{{ return "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256"; }}}
	};
```

# Recommendations

https://wiki.mozilla.org/Security/Server_Side_TLS

## Intermediate compatibility (recommended)

```
[root@icinga2-centos7-dev icinga2]# grep cipher_list /usr/local/icinga2/etc/icinga2/features-enabled/api.conf
  cipher_list = "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384"
```

```
$ openssl s_client -connect 192.168.33.21:5665
CONNECTED(00000003)
4647577196:error:14004410:SSL routines:CONNECT_CR_SRVR_HELLO:sslv3 alert handshake failure:/BuildRoot/Library/Caches/com.apple.xbs/Sources/libressl/libressl-22.260.1/libressl-2.6/ssl/ssl_pkt.c:1205:SSL alert number 40
4647577196:error:140040E5:SSL routines:CONNECT_CR_SRVR_HELLO:ssl handshake failure:/BuildRoot/Library/Caches/com.apple.xbs/Sources/libressl/libressl-22.260.1/libressl-2.6/ssl/ssl_pkt.c:585:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 7 bytes and written 0 bytes
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID:
    Session-ID-ctx:
    Master-Key:
    Start Time: 1564475774
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
---
```

## Old backward compatibility

```
[root@icinga2-centos7-dev icinga2]# grep cipher_list /usr/local/icinga2/etc/icinga2/features-enabled/api.conf
  cipher_list = "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:AES256-GCM-SHA384"
```

```
$ openssl s_client -connect 192.168.33.21:5665
CONNECTED(00000003)
depth=1 CN = Icinga CA
verify error:num=19:self signed certificate in certificate chain
verify return:0
---
Certificate chain
 0 s:/CN=icinga2-centos7-dev.vagrant.demo.icinga.com
   i:/CN=Icinga CA
 1 s:/CN=Icinga CA
   i:/CN=Icinga CA
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIFIDCCAwigAwIBAgIUGSW6kdl1gcKt/r981ZFNHB6uOFMwDQYJKoZIhvcNAQEL
BQAwFDESMBAGA1UEAwwJSWNpbmdhIENBMB4XDTE5MDYxODEyNDcwMloXDTM0MDYx
NDEyNDcwMlowNjE0MDIGA1UEAwwraWNpbmdhMi1jZW50b3M3LWRldi52YWdyYW50
LmRlbW8uaWNpbmdhLmNvbTCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoCggIB
AN9GZBhYIDTSMotz9vSqc9lq9bzFG63u1E8Tbj8nQspFPgL3TSS2xaDJPswy4lnw
82bemLrYCBYFlKbX9jhT6/MNJEXkxT3itcaNmtmekrZWzB5yYuTs6hjZTG78f/1n
LLoDmSmR441m0ize5o+dgqBNMWaM21MVFivRp+HBMrmuxjWk+jozQo8tc5ooSOpu
ZvbIXUXGnzkLad9dPMxEDVgQpnIEHM3NXscnl3xjasShAjBWw4TZB7BzVvdslkXJ
WmD+Zx3lDXKRBF8gSPwXBaZ/6zICw33AOsekgmj9yZ+D0lPIFsCZMOS0fr3jJJwC
//1lRVxuaGqiFNctVOH4mnk4HgMz2GDUX5Gg4wDjVbIdqlI/gMLWAvtOVIpGb6RS
ukM8PkGD5x18sKk6saJ4WL+BbBTcYJG0PeWbYgn5KrOkioZUAlBu7+Z3myAd+OQm
epeGR2j4dl9bWWm9zrb5DThl+rEP8DxNQlVPvFBLBqzo1tcXsvf+Epj0r5hWJZdt
qlNShsoZCuY+o95t1RZLYCNUzfhBfrutL87W71QX6iJSY+BqspNV5lcNV4ExET/0
/0jJhnPT263yY0mspdqt21eNcJHuTZXu3UQ4SwDQgEoV9Gp4ojvV7HRGryTJRyeN
LLkcJkjOE3WFXO8KVz+0DpnNS9vLrn1Zvj/J/b3F6A6BAgMBAAGjSDBGMAwGA1Ud
EwEB/wQCMAAwNgYDVR0RBC8wLYIraWNpbmdhMi1jZW50b3M3LWRldi52YWdyYW50
LmRlbW8uaWNpbmdhLmNvbTANBgkqhkiG9w0BAQsFAAOCAgEAdJJJpywVmf76LeRt
M16el9mQIzL/ckHQ7r0PTS2rDpdRwTGrt7NIhIjxaSI0UUtHZ/4iwExTAbaZg1LJ
+VmLsmfXRPPRfz4vND06tigCKVTd473R9Q29DAiFnyKzKdS0tctmJ+6BfDPZ0fG2
2VVub7KzACBNf/HbrCTj/iyKEri5MXjMQuK6Xf8NTFsmdf1TqeHdP3doMn2nDwT1
yyDtl4hA5MjOUPb0gd/aAVFoN6M/mJh0ZZRumDMSo7MaVWZNh3sXkpq1zfwWHf7H
7xRIEWg+b9WrnAn8hLjRLBHA3fK21topMQSRVtX15s3mNYHV3dA/62VgzbCkRD6U
0om52r/F3QXh7yugi/i3TKMVtpkC6nc5ma9wUIaztaPJGP0PJMQNrNiFteB+MTtQ
xQuz8yRoviRfyC0+K2wQMV8wt2QIlR3WpUon/oqyPq/HSc8/KfT0oOvn4P983mN8
GaOlGlO3e8N9WFNdsI0+Srum9TuJSgj0AIDBwA77csyvDhi8FPDEp5Po8emJX0rh
5aFrBs8qd//pVDJERuXnm5j5UAcXexZ+CVfSsBHtfBr2lIJBHvwCt73KqDgOguXi
E0RyfTh9hmoKt0HDrtMiPXYC+ZMyKCkPVsd8BowuzjR9mVCgiGLG46cV4KsSxPvw
snF6iQ5NanS4fuIsjCEp7u2nSgA=
-----END CERTIFICATE-----
subject=/CN=icinga2-centos7-dev.vagrant.demo.icinga.com
issuer=/CN=Icinga CA
---
Acceptable client certificate CA names
/CN=Icinga CA
---
SSL handshake has read 2923 bytes and written 786 bytes
---
New, TLSv1/SSLv3, Cipher is AES256-GCM-SHA384
Server public key is 4096 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : AES256-GCM-SHA384
    Session-ID: 6B09A49BB1471D9C043600CFC9E3310A1AF4A76B3D2FE9C7CDB3AFCA3C4C30F2
    Session-ID-ctx:
    Master-Key: 5CA5F7FB3ABCB18DC78CAF399A94BB76D00321157D35B8DB6CA1435E31D27840F84F6CF50BC8FEBCCC0B92EDBA73938F
    TLS session ticket lifetime hint: 300 (seconds)
    TLS session ticket:
    0000 - c8 bc e6 f0 2c 13 79 61-94 c4 5d c3 9d 60 20 42   ....,.ya..]..` B
    0010 - a1 46 01 c3 69 ac 47 4c-6e 66 58 19 c1 c2 59 5c   .F..i.GLnfX...Y\
    0020 - cb 14 d7 20 30 28 6a e9-62 34 a8 02 66 e9 73 d8   ... 0(j.b4..f.s.
    0030 - 70 ab 6d a5 e1 e8 64 6a-32 ea a9 96 ed a0 c5 6b   p.m...dj2......k
    0040 - dc ec 8f a5 09 65 33 ed-c0 9d 04 8a 0c c9 68 5f   .....e3.......h_
    0050 - 49 2d ca 5e 47 60 22 30-d7 f1 1b 3f db f6 e4 7d   I-.^G`"0...?...}
    0060 - fb af a7 12 d1 3a f0 01-3e aa a8 14 83 77 dd 43   .....:..>....w.C
    0070 - 70 7e 68 18 d3 87 f5 53-90 54 cd 6b ea 4d 29 53   p~h....S.T.k.M)S
    0080 - f2 53 39 35 58 6b c8 fd-4b ea 1d 30 5d f8 4c 3f   .S95Xk..K..0].L?
    0090 - 67 4b ff 52 d1 46 4c 9c-5a 74 39 d5 9a 91 ef 49   gK.R.FL.Zt9....I

    Start Time: 1564475892
    Timeout   : 7200 (sec)
    Verify return code: 19 (self signed certificate in certificate chain)
---
closed
```

**Comment by dnsmichi** [comment]
@peteeckel Agreed, upgrading to a newer version isn't the preferred solution by admins. Still, it should be done in preparation for future versions where compatibility ciphers may be dropped. And also for the request of elliptic curve certificates which is currently blocked with the missing ECDH cipher suites in el7 as well.

I've tested various ciphers with an el7 dev box (see above), and `AES256-GCM-SHA384` seems to be a good fit. This cipher will be only be selected when the satellite is 2.10 and receives a connection as `server`. #7368 adds this.

It may be a good idea to mention the "use 2.11 everywhere" in the upgrading or compatibility docs, allowing users to catch up.

**Comment by peteeckel** [comment]
@dnsmichi Perfect. 

As for upgrading the 2.10 agents, at least my practice is indeed to upgrade all agents to identical versions (notwithstanding version restrictions dictated by OS limitations), but you have to start somewhere and in some HA environments it's not very well taken when agents get disconnected during the upgrade ...

As for `AES256-GCM-SHA384`, that would have been my choice as well.

**Comment by dnsmichi** [comment]
@peteeckel thanks for your hard work on this, much appreciated. 

We hear you with binary upgrades, that thinking is more long term. We all have customers running on older versions. Considering the ECC draft, this likely only hits 2.13 or future major versions when 2.11 is "old stable" for agents. It may help to patch older versions though making the move more convenient for our users.

**Comment by dnsmichi** [comment]
- #7368 will be merged to master for 2.11, with a more compatible cipher list by default
- #7369 will be merged for 2.10.6 and can be cherry-picked into support/2.9 as well.

**Comment by dnsmichi** [comment]
Apart from the software changes involved, the more gentle approach is to soften the cipher list to not only support EDCH but also secure RSA ciphers.

The workaround with `cipher_list` inside the api feature sits inside the RC1 release blog post as update for quite a while, and also is explained in https://github.com/Icinga/icinga2/issues/7386
